Angular/my-app/src/app/task-list/task-list.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { items, Task } from '../tasks'
+
+@Component({
+  selector: 'app-task-list',
+  templateUrl: './task-list.component.html',
+  styleUrls: ['./task-list.component.css']
+})
+export class TaskListComponent implements OnInit {
+
+  constructor() { }
+  items =  items;
+  task: string;
+  description: string;
+
+  ngOnInit(): void {
+  }
+
+  go(id: number){
+    let i = this.items.indexOf(this.items.find(item => item.id == id));
+    let t: Task = this.items[i];
+    window.alert(t.task + ' ' + t.description + ' ' + t.done);
+  }
+
+  onCheck(id: number){
+    let i = this.items.indexOf(this.items.find(item => item.id == id));
+    console.log(this.items[i].done);
+    this.items[i].done = !this.items[i].done;
+    console.log(this.items[i].done);
+  }
+
+  addTask(task: string, description: string) : void{
+    if(task==null || task.trim()=="" || description==null || description.trim()=="")
```

Comment:
In TS you can use this:
```suggestion
    if(!task || !task.trim() || !description || !description.trim())
```